### PR TITLE
Add props to Hubspot Create Engagement Action

### LINKED ISSUE #6747: [FEATURE]
We should add two optional props for the Hubspot Create Engagement action:

- `toObjectId`
- `associationTypeId`

These fields are described in [this API Docs section](https://developers.hubspot.com/docs/api/crm/notes).

Reported by a user in Slack.

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #6747

### CODE REVIEW

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 1]
```diff
@@ -37,24 +41,95 @@ export default {
         },
       ],
     },
+    toObjectType: {
+      propDefinition: [
+        common.props.hubspot,
+        "objectType",
+      ],
+      label: "Associated Object Type",
+      description: "Type of object the from engagement is being associated with",
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:50]:
```suggestion
      description: "Type of object the engagement is being associated with",
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:50]: Changed "from" to "the" in the description to improve clarity.

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 2]
```diff
@@ -37,24 +41,95 @@ export default {
...
+    toObjectId: {
+      propDefinition: [
+        common.props.hubspot,
+        "objectId",
+        (c) => ({
+          objectType: c.toObjectType,
+        }),
+      ],
+      label: "Associated Object",
+      description: "Id of object the engagement is being associated with",
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:62]:
```suggestion
      description: "ID of object the engagement is being associated with",
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:62]: Changed "Id" to "ID" in the description to follow capitalization rules.

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 3]
```diff
@@ -37,24 +41,95 @@ export default {
...
+    associationType: {
+      propDefinition: [
+        common.props.hubspot,
+        "associationType",
+        (c) => ({
+          fromObjectType: c.engagementType,
+          toObjectType: c.toObjectType,
+        }),
+      ],
+      description: "A unique identifier to indicate the association type between the task and the other object",
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:74]:
```suggestion
      description: "A unique identifier to indicate the association type between the task and the other object",
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:74]: Changed "the" to "the other" in the description to improve clarity.

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 4]
```diff
@@ -37,24 +41,95 @@ export default {
...
       engagementType,
+      toObjectType,
+      toObjectId,
+      associationType,
       $db,
       ...properties
     } = this;
+
+    if ((toObjectId && !associationType) || (!toObjectId && associationType)) {
+      throw new ConfigurationError("Both `toObjectId` and `associationType` must be entered");
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:111]:
```suggestion
      throw new ConfigurationError("Both `toObjectId` and `associationType` must be entered");
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:111]: Changed "entered" to "provided" in the error message to improve clarity.

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 5]
```diff
@@ -37,24 +41,95 @@ export default {
         },
       ],
     },
+    toObjectType: {
+      propDefinition: [
+        common.props.hubspot,
+        "objectType",
+      ],
+      label: "Associated Object Type",
+      description: "Type of object the engagement is being associated with",
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:50]:
```suggestion
      description: "Type of object the engagement is associated with",
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:50]: Changed "being associated with" to "associated with" to follow Google's Developer docs style guide (https://developers.google.com/style/voice)

File: components/hubspot/actions/create-engagement/create-engagement.mjs [hunk 6]
```diff
@@ -37,24 +41,95 @@ export default {
...
+    toObjectId: {
+      propDefinition: [
+        common.props.hubspot,
+        "objectId",
+        (c) => ({
+          objectType: c.toObjectType,
+        }),
+      ],
+      label: "Associated Object",
+      description: "ID of object the engagement is being associated with",
```
Comment (dylburger) [inline components/hubspot/actions/create-engagement/create-engagement.mjs:62]:
```suggestion
      description: "ID of object the engagement is associated with",
```

Comment (dylburger) [reply components/hubspot/actions/create-engagement/create-engagement.mjs:62]: Changed "being associated with" to "associated with" to follow Google's Developer docs style guide (https://developers.google.com/style/voice)

**Review by vellames — APPROVED** [review]
lgtm

**Comment by vunguyenhung** [comment]
Hi, I confirmed that the new change is good. Ready for release!
image
image

**Pull request merged**